### DISCUSSION TITLE
Is there a way to add "world info" cards to a session, persistent or otherwise?

### DISCUSSION THREAD

**Discussion by FrankDMartinez** [ggml-org/llama.cpp#2150] [Q&A]
I know KoboldCPP allows for "world info" cards to be added for new information, such as if I want to update the name of the Prime Minister. However, I was unable to find an equivalent for LLaMa.cpp. Any suggestions?

Thanks in advance.

**Comment by KerfuffleV2** [comment]
It wouldn't be doing anything special. It would be some extra information information in the prompt submitted to the LLM. If you can find out the format KoboldCPP uses, you could replicate what it does pretty easily.

If you want to avoid having to repeatedly evaluate that part of the prompt (presumably it won't change often) you could look into the prompt caching options.

A super simple example of what I'm talking about (with Alpaca style prompting)

```plaintext
Keep the following information in mind when answering questions:
The current Prime Minister of England is a German Shepherd named Rover.

### Instruction: Who is the current Prime Minister of England?
### Response:
```

And presumably you'll get "Rover" as the answer. Depending on the model, the prompt format/method for supplying additional information may need tweaking.
